tests/models/test_evaluation.py

```diff
@@ -333,7 +336,33 @@ def iris_pandas_df_num_cols_dataset():
     eval_X, eval_y = X[0::3], y[0::3]
     data = pd.DataFrame(eval_X)
     data["y"] = eval_y
-    return EvaluationDataset(data=data, targets="y", name="iris_pandas_df_num_cols_dataset")
+    constructor_args = {"data": data, "targets": "y", "name": "iris_pandas_df_num_cols_dataset"}
+    ds = EvaluationDataset(**constructor_args)
+    ds._constructor_args = constructor_args
+    return ds
+
+
+@pytest.fixture
+def iris_pandas_df_longname_cols_dataset():
+    X, y = get_iris()
+    eval_X, eval_y = X[0::3], y[0::3]
+    data = pd.DataFrame(
+        {
+            "f1": eval_X[:, 0],
+            "f2": eval_X[:, 1],
+            "f3longnamelongnamelongname": eval_X[:, 2],
+            "f4longnamelongnamelongname": eval_X[:, 3],
```

Comment:
```suggestion
            # Column names longer than the truncation threshold
            "f3_" + "x" * 20: eval_X[:, 2],
            "f4_" + "x" * 20: eval_X[:, 3],
```

I'd use `+ "x" * 20` here. It's a bit hard to tell if `f3longnamelongnamelongname` has more than 20 characters.